### PR TITLE
Resolve Lightning App with remote storage

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Currently, accessing the Drive or Path doesn't work. This PR should address it. 

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca

### CODE REVIEW

File: src/lightning/app/core/constants.py [hunk 1]
```diff
@@ -20,6 +20,9 @@
 
 
 def get_lightning_cloud_url() -> str:
+    # detect local development
+    if os.getenv("VSCODE_PROXY_URI", "").startswith("http://localhost:9800"):
+        return "http://localhost:9800"
```
Comment (awaelchli) [inline src/lightning/app/core/constants.py:25]: Are you sure you want to include the hardcoded port number?

**Review by awaelchli — APPROVED** [review]

File: src/lightning/app/storage/path.py [hunk 2]
```diff
@@ -427,16 +427,7 @@ def _filesystem() -> AbstractFileSystem:
...
-        secret = os.getenv("LIGHTNING_AWS_SECRET_ACCESS_KEY", "")
-        # TODO: Remove when updated on the platform side.
-        if key == "" or secret == "":
-            key = os.getenv("AWS_ACCESS_KEY_ID", "")
-            secret = os.getenv("AWS_SECRET_ACCESS_KEY", "")
-        if key == "" or secret == "":
-            raise RuntimeError("missing S3 bucket credentials")
-
-        fs = S3FileSystem(key=key, secret=secret, use_ssl=False, client_kwargs={"endpoint_url": endpoint_url})
+        fs = S3FileSystem()
```
Comment (ethanwharris) [inline src/lightning/app/storage/path.py:430]: We will need this here too I think:
```
del os.environ["AWS_ACCESS_KEY_ID"]
del os.environ["AWS_SECRET_ACCESS_KEY"]
```

Comment (tchaton) [reply src/lightning/app/storage/path.py:430]: I will prefer not as users might pass their own.

File: src/lightning/app/core/constants.py (hunk 1 above)
Comment (tchaton) [reply src/lightning/app/core/constants.py:25]: Yes, it is fine for now. Thanks for asking :)

**Review by ethanwharris — APPROVED** [review]

File: src/lightning/app/storage/path.py (hunk 2 above)
Comment (tchaton) [reply src/lightning/app/storage/path.py:434]: Done ! Let's sequence this with 2 PR.

File: src/lightning/app/testing/testing.py [hunk 3]
```diff
@@ -385,6 +385,8 @@ def run_app_in_cloud(
 
             # wait until the app is running and openapi.json is ready
             if app.status.phase == V1LightningappInstanceState.RUNNING:
+                break
```
Comment (Borda) [inline src/lightning/app/testing/testing.py:388]: pending on this resolution

**Review by Borda — CHANGES_REQUESTED** [review]

File: src/lightning/app/testing/testing.py (hunk 3 above)
Comment (tchaton) [reply src/lightning/app/testing/testing.py]: Done !

File: .azure/app-cloud-e2e.yml [hunk 4]
```diff
@@ -67,57 +67,42 @@ jobs:
         'App: v0_app':
           name: "v0_app"
           dir: "public"
-          queue_type: "redis"
```
Comment (Borda) [inline .azure/app-cloud-e2e.yml:70]: @ethanwharris why do we drop redis?

File: examples/app/commands_and_api/.lightningignore [hunk 5]
```diff
@@ -0,0 +1 @@
+venv/
```
Comment (Borda) [inline examples/app/commands_and_api/.lightningignore:1]: is this used?

File: src/lightning/app/core/constants.py [hunk 6]
```diff
@@ -115,17 +118,5 @@ def enable_interruptible_works() -> bool:
...
-    value = os.getenv("LIGHTNING_CLUSTER_DRIVER", None)
-    if value is None:
-        if enable_interruptible_works():
-            value = "direct"
-        else:
-            value = None
-    if value not in _CLUSTER_DRIVERS:
-        raise ValueError(f"Found {value} cluster driver. The value needs to be in {_CLUSTER_DRIVERS}.")
-    return value
+    return "direct"
```
Comment (Borda) [inline src/lightning/app/core/constants.py:122]: why unset `LIGHTNING_CLUSTER_DRIVER` is not enough?

File: src/lightning/app/core/queues.py [hunk 7]
```diff
@@ -388,19 +388,30 @@ def get(self, timeout: Optional[float] = None) -> Any:
         if timeout is None:
             while True:
                 try:
-                    return self._get()
+                    try:
```
Comment (Borda) [inline src/lightning/app/core/queues.py:391]: let's use `contextmanager.suppress(...)`

File: src/lightning/app/core/queues.py [hunk 8]
```diff
@@ -388,19 +388,30 @@ def get(self, timeout: Optional[float] = None) -> Any:
...
+                        pass
                 except queue.Empty:
                     time.sleep(HTTP_QUEUE_REFRESH_INTERVAL)
 
         # make one request and return the result
         if timeout == 0:
-            return self._get()
+            try:
+                return self._get()
+            except requests.exceptions.HTTPError:
```
Comment (Borda) [inline src/lightning/app/core/queues.py:402]: can you pls add a note about why this error happens and it is fine to ignore it?

File: tests/integrations_app/public/test_v0_app.py [hunk 9]
```diff
@@ -53,6 +53,7 @@ def check_content(button_name, text_content):
     has_logs = False
     while not has_logs:
         for log in fetch_logs(["flow"]):
+            print(log)
```
Comment (Borda) [inline tests/integrations_app/public/test_v0_app.py:56]: remove?

File: tests/tests_app/core/test_constants.py [hunk 10]
```diff
@@ -1,25 +1,9 @@
...
-from lightning.app.core.constants import get_cluster_driver, get_lightning_cloud_url
+from lightning.app.core.constants import get_lightning_cloud_url
 
 
 @mock.patch.dict(os.environ, {"LIGHTNING_CLOUD_URL": "https://beta.lightning.ai"})
 def test_defaults():
     assert get_lightning_cloud_url() == "https://beta.lightning.ai"
-
-
-def test_cluster_drive(monkeypatch):
```
Comment (Borda) [inline tests/tests_app/core/test_constants.py:14]: so this is not expected to work?

File: .azure/app-cloud-e2e.yml (hunk 4 above)
Comment (tchaton) [reply .azure/app-cloud-e2e.yml:70]: Yes, this won't be supported going forward.

File: examples/app/commands_and_api/.lightningignore (hunk 5 above)
Comment (tchaton) [reply examples/app/commands_and_api/.lightningignore:1]: This is automatically generated when the app runs. Better to have it in the repo.

File: src/lightning/app/core/constants.py (hunk 6 above)
Comment (tchaton) [reply src/lightning/app/core/constants.py:122]: This isn't used and we can drop it entirely.

Comment (tchaton) [reply src/lightning/app/core/constants.py:122]: For another PR once the fire is being us.

File: tests/integrations_app/public/test_v0_app.py (hunk 9 above)
Comment (tchaton) [reply tests/integrations_app/public/test_v0_app.py:56]: Good catch !

**Pull request merged**